src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerProfileUpdateService.kt

```diff
@@ -313,6 +324,173 @@ class PrisonerProfileUpdateService(
     ethnicity = ethnicity?.toReferenceDataValue(),
   )
 
+  @Transactional
+  fun getCommunicationNeeds(prisonerNumber: String): CorePersonCommunicationNeeds {
+    try {
+      val booking = offenderBookingRepository.findLatestOffenderBookingByNomsId(prisonerNumber)
+        .orElseThrowNotFound("Prisoner with prisonerNumber %s and existing booking not found", prisonerNumber)
+
+      val languages = offenderLanguageRepository.findByOffenderBookId(booking.bookingId)
+
+      return CorePersonCommunicationNeeds(
+        prisonerNumber,
+        languagePreferences = CorePersonLanguagePreferences(
+          preferredSpokenLanguage = getFirstPreferredSpokenLanguage(languages)?.referenceCode,
+          preferredWrittenLanguage = getFirstPreferredWrittenLanguage(languages)?.referenceCode,
+          interpreterRequired = getInterpreterRequired(languages),
+        ),
+        secondaryLanguages = getSecondaryLanguages(languages),
+      )
+    } catch (e: CannotAcquireLockException) {
```

Comment:
I don't think anything here does a pessimistic lock so it won't throw this exception.